### PR TITLE
Cherry-pick big number changes and code cleanup

### PR DESCRIPTION
**Pull request by kristw**
Cherry-pick the following commits

SHA | Time | Log | Author
--------  | ------------ | ------------- | --------------
39acd9ff | Mon Aug 6 17:26:06 2018 -0700 | remove deep-equal and use _.isEqual instead (#5573) | [Krist Wongsuphasawat]
aa14bac5 | Mon Aug 6 14:32:29 2018 -0700 | Add option for BigNumber to not start y-axis at 0 (#5552) | [Krist Wongsuphasawat]
9d95c4cc | Thu Aug 2 11:45:58 2018 -0700 | remove js packages that are not referenced in js code (#5514) | [Krist Wongsuphasawat]
1b9e5d41 | Thu Aug 2 10:34:36 2018 -0700 | Update BigNumber design (#5469) | [Krist Wongsuphasawat]
7158fb1e | Fri Jul 6 23:09:25 2018 +0900 | added a 'no_trend_line' option (#5356) | [Yu Xiao]

@graceguo-supercat @michellethomas @john-bodley @williaster

### CODE REVIEW

File: superset/assets/src/visualizations/big_number.js [hunk 1]
```diff
@@ -93,23 +93,24 @@ function bigNumberVis(slice, payload) {
 
   if (fd.viz_type === 'big_number') {
```
Comment (williaster) [inline superset/assets/src/visualizations/big_number.js]: should this file still be around?

File: yarn.lock [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (williaster) [inline yarn.lock]: should this file be here?

File: superset/assets/src/visualizations/big_number.js (hunk 1 above)
Comment (kristw) [reply superset/assets/src/visualizations/big_number.js]: Good catch. The merge must did something wrong.

File: yarn.lock (hunk 2 above)
Comment (kristw) [reply yarn.lock]: Will remove and re-push

**Pull request merged**